functions.md

```diff
@@ -0,0 +1,131 @@
+
+```js
+function monitorCount(rows, columns) {
+   let total = rows + columns
+}
+console.log(monitorCount(3 + 5));// print undefined
+```
+
+ the `return` keyword allow us see the calculation that the computer did print example:
+
+```js 
+function totalNumber(cash, bank) {
+    return 30 +80;
+}
+console.log(totalNumber()); // 11
+```
+
+The `return` keyword is powerful because it allows functions to produce an output. We can then save the output to a variable for later use
+
+### Function Expression
+ Another way of defining a function is to use a function expression. example:
+ 
+ ```js
+ const myName = Function(boy {
+
+ })
+  ```
+  To declare a function expression:
+
+- Declare a variable to make the variable’s name be the name, or identifier, of your function. Since the release of ES6, it is common practice to use const as the keyword to declare the variable.
+
+- Assign as that variable’s value an anonymous function created by using the function keyword followed by a set of parentheses with possible parameters. Then a set of curly braces that contain the function body. Unlike function declarations, function expressions are not hoisted so they cannot be called before they are defined.
+
+  ### Arrow Function
+  arrow function is another way to deefine function introduced by the `ES6` It's a shorter way to write function using the `=>` arrow. example:
+
+  ```js
+  const plantNeedWater = (day) => {
+      if (day === 'Monday') {
+          return true;
+      }
+  }
+  ```
+
+It’s important to be familiar with the multiple ways of writing functions because you will come across each of these when reading other JavaScript code.
+
+### Concise Body Arrow Functions
+
+JavaScript also provides several ways to refactor arrow function syntax. The most condensed form of the function is known as concise body.
+
+- Functions that take only a single parameter do not need that parameter to be enclosed in parentheses. However, if a function takes zero or multiple parameters, parentheses are required.
+
+- showcasing how arrow functions parameters differ for different amounts of parameters
+A function body composed of a single-line block does not need curly braces. Without the curly braces, whatever that line evaluates will be automatically returned. The contents of the block should immediately follow the arrow => and the return keyword can be removed. This is referred to as implicit return.
+
+So if we have a function:
+
+```js
+const squareNum = (num) => {
+  return num * num;
+};
+```
+
+We can refactor the function to:
+
+```js    
+const squareNum = num => num * num;
+```
+
+Notice the following changes:
+
+- The parentheses around num have been removed, since it has a single parameter.
+- The curly braces { } have been removed since the function consists of a single-line block.
+- The return keyword has been removed since the function consists of a single-line block.
```

Comment:
As suggested in other reviews, add resources used, and an example that shows the usages of all you have been explaining so far.